src/Components/Components/src/Reflection/MemberAssignment.cs

```diff
@@ -11,18 +12,23 @@ namespace Microsoft.AspNetCore.Components.Reflection
     internal class MemberAssignment
     {
         public static IEnumerable<PropertyInfo> GetPropertiesIncludingInherited(
-            Type type, BindingFlags bindingFlags)
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.NonPublicProperties)] Type type,
+            BindingFlags bindingFlags)
```

Comment:
I'm not sure this is enough. The current behavior in linker is that it follow reflection exactly. So this annotation is equivalent to calling type.GetProperties(bindingflags). It does NOT translate to calling the same on the base type. Specifically, private properties on base type are not preserved, but the code here will try to access them.

The reason there's no warning is that linker has a known issue with tracking backward branches, and so the while loop is not correctly analyzed (it sees the `.BaseType` "too late" and doesn't realize that it will call `.GetProperties()` on it).

I think the only "safe" annotation is `All` in this case (which does preserve everything, including private properties on all base types).

Side note: Why do we allow private reflection here... we're telling everybody not to do it, and here we allow it.